File: api/db/enum/stratagem_type.go
package enum

import (
	"database/sql/driver"
	"fmt"
)

type StratagemType string

const (
	Supply    StratagemType = "supply"
	Mission   StratagemType = "mission"
	Defensive StratagemType = "defensive"
	Offensive StratagemType = "offensive"
)

func (st *StratagemType) Scan(value interface{}) error {
	b, ok := value.([]byte)
	if !ok {
		switch value.(string) {
		case string(Supply), string(Mission), string(Defensive), string(Offensive):
			*st = StratagemType(b)
		default:
			return fmt.Errorf("invalid value for StratagemType: %v", value)
		}
	}
	return nil
}

func (st StratagemType) Value() (driver.Value, error) {
	return string(st), nil
}

File: mobile/lib/enum/stratagem_type.dart
import 'package:flutter/material.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:flutter_gen/gen_l10n/app_localizations.dart';

enum StratagemType {
  @JsonValue('supply')
  supply(
    'supply',
  ),

  @JsonValue('mission')
  mission(
    'mission',
  ),

  @JsonValue('defensive')
  defensive(
    'defensive',
  ),

  @JsonValue('offensive')
  offensive(
    'offensive',
  );

  const StratagemType(
    this.code,
  );

  /// The code of the difficulty. Use [translatedName] to get the translated name.
  final String code;

  String translatedName(BuildContext context) {
    switch (this) {
      case StratagemType.supply:
        return AppLocalizations.of(context)!.stratagemTypeSupply;
      case StratagemType.mission:
        return AppLocalizations.of(context)!.stratagemTypeMission;
      case StratagemType.defensive:
        return AppLocalizations.of(context)!.stratagemTypeDefensive;
      case StratagemType.offensive:
        return AppLocalizations.of(context)!.stratagemTypeOffensive;
    }
  }
}
